File: SeatbeltRule.java
import java.util.Optional;

public class SeatbeltRule implements Rule {
    private static final String RULE_NAME = "Seatbelt Rule";
    private final double fee;

    public SeatbeltRule(double fee) {
        this.fee = fee;
    }

    public SeatbeltRule() {
        this(100.0);
    }

    @Override
    public String getRuleName() {
        return RULE_NAME;
    }

    @Override
    public Optional<Violation> evaluate(RadarObservation observation) {
        if (!observation.isSeatbeltFastened()) {
            return Optional.of(new Violation(
                RULE_NAME,
                "Seatbelt not fastned",
                fee
            ));
        }
        return Optional.empty();
    }
}

File: python version/seatbelt_rule.py
from typing import Optional
from rule import Rule
from radar_observation import RadarObservation
from violation import Violation

class SeatbeltRule(Rule):
    RULE_NAME = "Seatbelt Rule"

    def __init__(self, fee: float = 100.0):
        self.fee = fee

    def get_rule_name(self) -> str:
        return self.RULE_NAME

    def evaluate(self, observation: RadarObservation) -> Optional[Violation]:
        if not observation.seatbelt_fastened:
            return Violation(self.RULE_NAME, "Seatbelt not fastned", self.fee)
        return None
